modules/imap/handler_modules.php

```diff
@@ -924,6 +924,86 @@ public function process() {
     }
 }
 
+/**
+ * Snooze message
+ * @subpackage imap/handler
+ */
+class Hm_Handler_imap_snooze_message extends Hm_Handler_Module {
+    /**
+     * Use IMAP to snooze the selected message uid
+     */
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_snooze_ids', 'imap_snooze_until'));
+        if (!$success) {
+            return;
+        }
+        $snoozed_messages = 0;
+        $snooze_tag = null;
+        if ($form['imap_snooze_until'] != 'unsnooze') {
+            $at = date('D, d M Y H:i:s O');
+            $until = get_snooze_date($form['imap_snooze_until']);
+            $snooze_tag = "X-Snoozed: at $at;\n \tuntil $until";
```

Comment:
Why two line headers? This might cause issues in some MTAs or MUAs. Just use a single line - you don't need the new line here to be able to parse it later.